packages/core/src/internal/base/button.base.ts

```diff
@@ -104,26 +119,48 @@ export class CdsBaseButton extends LitElement {
     if (!this.readonly) {
       if (this.disabled) {
         stopEvent(event);
-      } else if (!event.defaultPrevented) {
-        const buttonTemplate = html`<button
-          aria-hidden="true"
-          ?disabled="${this.disabled}"
-          tabindex="-1"
-          style="display: none !important"
-          value="${ifDefined(this.value)}"
-          name="${ifDefined(this.name)}"
-          type="${ifDefined(this.type)}"
-        ></button>`;
-
-        const marker = document.createElement('div');
-        this.appendChild(marker);
-        render(buttonTemplate, marker);
+      } else if (!event.defaultPrevented && this.marker && this.explicitClick) {
+        if (this.markerAttached()) {
+          this.appendChild(this.marker);
+        }
         this.querySelector('button[aria-hidden]')?.dispatchEvent(
           new MouseEvent('click', { relatedTarget: this, composed: true })
         );
-        marker.remove();
+        if (this.type !== 'submit') {
+          this.marker.remove();
+        }
       }
     }
+    // get ready for the next real click check
+    this.explicitClick = false;
+  }
+
+  private createMarker() {
+    const buttonTemplate = html`<button
+      aria-hidden="true"
+      role="presentation"
+      ?disabled="${this.disabled}"
+      tabindex="-1"
+      style="display: none !important"
+      value="${ifDefined(this.value)}"
+      name="${ifDefined(this.name)}"
+      type="${ifDefined(this.type)}"
+    ></button>`;
+
+    const marker = document.createElement('div');
+    // avoid an empty space inside button
+    marker.style.display = 'none';
+    render(buttonTemplate, marker);
+    this.marker = marker;
+  }
+
+  private markerAttached(): boolean {
```

Comment:
any reason why this is a method and not a getter?

Reply:
My understanding is that a property would most often reveal a state, attaching certain behaviors to setters/getters (especially setters), while functions bring some more complex logic along.
In this case, you are right - it could very well be a getter. Will be changed.

Reply:
done